crates/runtime/src/vmcalls/transfer.rs

```diff
@@ -0,0 +1,46 @@
+use crate::FuncEnv;
+use svm_state::AccountStorage;
+use svm_types::{Address, BytesPrimitive};
+
+fn load_addr(env: &FuncEnv, mem_ptr: u32) -> Address {
```

Comment:
I'd extract the code that reads bytes out of memory using const generics and have the `load_addr` code receive `[Address::len(); u8]` and only make an `Address.`

IIRC there's some API for that built-in in Wasmer (for grabbing a Wasm memory bytes)